cardano-node/src/Cardano/Tracing/OrphanInstances/Network.hs

```diff
@@ -1756,10 +1756,40 @@ instance Show vNumber => ToJSON (HandshakeException vNumber) where
                  ]
 
 instance ToJSON NodeToNodeVersion where
-  toJSON x = String (pack $ show x)
+  toJSON NodeToNodeV_7  = Number 7
+  toJSON NodeToNodeV_8  = Number 8
+  toJSON NodeToNodeV_9  = Number 9
+  toJSON NodeToNodeV_10 = Number 10
+  toJSON NodeToNodeV_11  = Number 11
+
+instance FromJSON NodeToNodeVersion where
+  parseJSON (Number 7) = return NodeToNodeV_7
+  parseJSON (Number 8) = return NodeToNodeV_8
+  parseJSON (Number 9) = return NodeToNodeV_9
+  parseJSON (Number 10) = return NodeToNodeV_10
+  parseJSON (Number 11) = return NodeToNodeV_11
+  parseJSON (Number x) = fail ("FromJSON.NodeToNodeVersion: unsupported node-to-node protocol version " ++ show x)
+  parseJSON _          = fail "FromJSON.NodeToNodeVersion: error parsing NodeToNodeVersion"
```

Comment:
Please `show` the JSON object it failed to parse in the error message.

Reply:
Done.